addons/calendar/models/calendar_alarm.py

```diff
@@ -55,10 +56,13 @@ def _search_duration_minutes(self, operator, value):
             '&', ('interval', '=', 'days'), ('duration', operator, value / 60 / 24),
         ]
 
-    @api.onchange('duration', 'interval', 'alarm_type')
+    @api.onchange('duration', 'interval', 'alarm_type', 'notify_responsible')
     def _onchange_duration_interval(self):
         display_interval = self._interval_selection.get(self.interval, '')
         display_alarm_type = {
             key: value for key, value in self._fields['alarm_type']._description_selection(self.env)
         }[self.alarm_type]
-        self.name = "%s - %s %s" % (display_alarm_type, self.duration, display_interval)
+        if self.alarm_type not in ('sms', 'whatsapp'):
+            self.notify_responsible = False
+        notify_text = f"- {_('Notify Responsible')}" if self.notify_responsible else ""
+        self.name = "%s - %s %s %s" % (display_alarm_type, self.duration, display_interval, notify_text)
```

Comment:
As we have `notify_responsible` default False, can't we do something like
```suggestion
       self.name = "%s - %s %s" % (display_alarm_type, self.duration, display_interval)
       if self.notify_responsible:
          self.name += " - " + _("Notify Responsible")
```